### PR TITLE
Inject migration plugin manager

### PR DESCRIPTION
**Pull request by claudiu-cristea**
Includes tightening typing and some cosmetic changes

### CODE REVIEW

File: src/Drupal/Commands/core/drush.services.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ services:
         -  { name: drush.command }
   migrate_runner.commands:
     class: Drush\Drupal\Commands\core\MigrateRunnerCommands
-    arguments: ['@date.formatter', '@keyvalue']
+    arguments: ['@date.formatter', '@keyvalue', '@?plugin.manager.migration']
```
Comment (weitzman) [inline src/Drupal/Commands/core/drush.services.yml:55]: I forget what this `@?` syntax means. Please add a comment.

File: src/Drupal/Migrate/MigrateMissingSourceRowsEvent.php [hunk 2]
```diff
@@ -3,7 +3,7 @@
 namespace Drush\Drupal\Migrate;
 
 use Drupal\migrate\Plugin\MigrationInterface;
-use Symfony\Component\EventDispatcher\Event;
+use Symfony\Contracts\EventDispatcher\Event;
```
Comment (weitzman) [inline src/Drupal/Migrate/MigrateMissingSourceRowsEvent.php:6]: Is 'contracts' OK for Symfony 4 and 5?

**Review by weitzman — APPROVED** [review]

File: src/Drupal/Commands/core/drush.services.yml (hunk 1 above)
Comment (claudiu-cristea) [reply src/Drupal/Commands/core/drush.services.yml:55]: If the service doesn't exists (e.g. the service module is not installed) will pass `null`. I would not add a comment in the YAML as this a very known syntax and we're already using it in Drush, see https://github.com/drush-ops/drush/blob/11.x/src/Drupal/Commands/field/drush.services.yml

Comment (weitzman) [reply src/Drupal/Commands/core/drush.services.yml:55]: Ah right, sounds good

File: src/Drupal/Migrate/MigrateMissingSourceRowsEvent.php (hunk 2 above)
Comment (claudiu-cristea) [reply src/Drupal/Migrate/MigrateMissingSourceRowsEvent.php:6]: Symfony 5.4 requires `"symfony/event-dispatcher-contracts": "^2|^3"`
Symfony 4.4 requires `"symfony/event-dispatcher-contracts": "^1.1"`

All `symfony/event-dispatcher-contracts` versions are exposing the Event class with the same namespace. See:

* https://github.com/symfony/event-dispatcher-contracts/blob/1.1/Event.php
* https://github.com/symfony/event-dispatcher-contracts/blob/2.3/Event.php
* https://github.com/symfony/event-dispatcher-contracts/blob/3.0/Event.php

Right now Drupal 9 doesn't allow to bump Symfony to 5, so we can't test. Anyway, maybe we should consider to run at least one test with `composer update --prefer-lowest` to check the lowest dependency versions scenario,

**Pull request merged**